src/lib.rs

```diff
@@ -570,6 +571,10 @@ where
     ///
     /// The return value is a result indicating whether the new value was written and containing
     /// the previous value. On success this value is guaranteed to be equal to `current`.
+    ///
+    /// The return value is a result indicating whether the new value was written successfully.
+    /// - On success the result contains the pointer previously in `self`, or None if it was null
+    /// - On failure the result gives back `new`
```

Comment:
This actually makes me think that we should update the `Err` case to also include the pointer value in `self` since it was `!= current` and callers will want to avoid having to do a second (unnecessary) `load`. That's a breaking change that should be made separately though.